src/engine/pieces/pawn.ts

```diff
@@ -1,13 +1,28 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square from '../square';
 
 export default class Pawn extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+
+        let moveArr: Square[] = []; // Array of squares to move to
```

Comment:
Similarly to my below comment, you might consider `validMoves` as an alternative name here; the fact that this is an array is encoded by the type, so it's redundant!